### PR TITLE
feat: Add suffix 'config' to config filename

### PR DESCRIPTION
**Pull request by regseb**
This PR adds the `stryker.config.{json,js,mjs,cjs}` and `.stryker.config.{json,js,mjs,cjs}` in the default config file names.

---

For configuration files, the extension `.config.js` is often used. The extension `.conf.js` is less common. Here are examples of different projects that use `.config.js`: [`eslint.config.js`](https://eslint.org/docs/latest/use/configure/configuration-files-new#configuration-file), [`prettier.config.{js,cjs}`](https://prettier.io/docs/en/configuration.html) (via [Cosmiconfig](https://github.com/davidtheclark/cosmiconfig#cosmiconfig)), [`jest.config.{js,ts,mjs,cjs,json}`](https://jestjs.io/docs/configuration/), [`nyc.config.js`](https://www.npmjs.com/package/nyc#configuration-files), [`webpack.config.js`](https://webpack.js.org/configuration/), [`vue.config.js`](https://cli.vuejs.org/config/#vue-config-js), [`next.config.{js,mjs}`](https://nextjs.org/docs/api-reference/next.config.js/introduction), [`docusaurus.config.js`](https://docusaurus.io/docs/api/docusaurus-config), [`stylelint.config.{cjs,js}`](https://stylelint.io/user-guide/configure), [`tailwind.config.js`](https://tailwindcss.com/docs/configuration) (via [Lilconfig](https://github.com/antonk52/lilconfig/blob/b056bd8e9d78eb258992005eedc855256af8bbe6/src/index.ts#L54-L55)), [`typedoc.config.{js,cjs}`](https://typedoc.org/options/configuration/#options).

| | `".config.js"` | `".conf.js"` |
| --- | --- | --- |
| GitHub | [_Code: 2.6M_](https://github.com/search?q=%22.config.js%22&type=code) | [_Code: 446k_](https://github.com/search?q=%22.conf.js%22&type=code) |
| npm | [_98 packages found_](https://www.npmjs.com/search?q=%22.config.js%22) | [_5 packages found_](https://www.npmjs.com/search?q=%22.conf.js%22) |
| Google Search | [_About 4,670,000 results_](https://www.google.com/search?q=".config.js"&hl=en) | [_About 232,000 results_](https://www.google.com/search?q=".conf.js"&hl=en) |
| Google Trends | [_Average: 32_](https://trends.google.fr/trends/explore?q=%22config.js%22,%22conf.js%22) | [_Average: 7_](https://trends.google.fr/trends/explore?q=%22config.js%22,%22conf.js%22) |

I propose this pull request because when creating a new project, I named my configuration files `*.config.js` (`eslint.config.js`, `stylelint.config.js`...). But when running Stryker, it couldn't find its configuration file, even though the `.stryker.config.js` file was present in the root of my project 🤔.

### CODE REVIEW

**Review by nicojs — CHANGES_REQUESTED** [review]
I agree that `.config.js` is the better suffix. The `.conf.js` suffix is one of the last remnants of our karma past (we used to support only karma, and karma uses karma.conf.js).

- Should we also generate `stryker.config.mjs` and `stryker.config.json` by default during `stryker init`?
- Could you change 1 or more e2e tests to use `stryker.config.js`?
- Do you also want to create the PR to include `stryker.config.[mjs,cjs,js,json]` to https://github.com/vscode-icons/vscode-icons?

**Comment by regseb** [comment]
I've pushed a new commit https://github.com/regseb/stryker-js/commit/1819a2e8f758de5688f548e80ea4a8f3640ea58f, but it doesn't display in this pull request.

- https://github.com/stryker-mutator/stryker-js/compare/master...regseb:stryker-js:config: _**2 commits** / 24 files changed / 1 contributor_
- https://github.com/regseb/stryker-js/tree/config: _This branch is [**2 commits ahead**](https://github.com/regseb/stryker-js/compare/stryker-mutator:stryker-js:master...config), [7 commits behind](https://github.com/regseb/stryker-js/compare/config...stryker-mutator:stryker-js:master) stryker-mutator:master._

I don't know why it doesn't add in this pull request?

---

I create pull request in https://github.com/vscode-icons/vscode-icons/pull/3309

File: docs/config-file.md [hunk 1]
```diff
@@ -50,34 +50,30 @@ You can use your editor's autocompletion to help you author your configuration f
 ## Usage
```
Comment (nicojs) [inline docs/config-file.md:51]:
```suggestion
<details>

<summary>History</summary>

| Version | Changes                     |
| ------- | --------------------------- |
| 7.1     | Add `.config.{json,js,mjs,cjs}` as default suffix. |

</details>
```

**Review by nicojs — APPROVED** [review]
I think it looks great so far.

However, I wonder if I would want to merge this before https://github.com/vscode-icons/vscode-icons/pull/3309 is resolved. I would hate for users to have to look at the default JSON icon, when they have the fancy Stryker icon now.

**Comment by robertohuertasm** [comment]
> However, I wonder if I would want to merge this before [vscode-icons/vscode-icons#3309](https://github.com/vscode-icons/vscode-icons/pull/3309) is resolved. I would hate for users to have to look at the default JSON icon, when they have the fancy Stryker icon now.

FYI, [v12.5.0](https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons) has been already published.

**Review by nicojs — APPROVED** [review]

**Pull request merged**